node/sovereignNode_test.go

```diff
@@ -0,0 +1,354 @@
+package node_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-core-go/data/api"
+	"github.com/multiversx/mx-chain-go/common"
+	"github.com/multiversx/mx-chain-go/node"
+	"github.com/multiversx/mx-chain-go/node/mock"
+	"github.com/multiversx/mx-chain-go/state"
+	"github.com/multiversx/mx-chain-go/testscommon"
+	stateMock "github.com/multiversx/mx-chain-go/testscommon/state"
+	trieMock "github.com/multiversx/mx-chain-go/testscommon/trie"
+	"github.com/multiversx/mx-chain-go/vm/systemSmartContracts"
+	vmcommon "github.com/multiversx/mx-chain-vm-common-go"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewSovereignNode(t *testing.T) {
+	n, err := node.NewNode()
+	require.Nil(t, err)
+	require.NotNil(t, n)
+
+	sn, err := node.NewSovereignNode(n)
+	require.Nil(t, err)
+	require.NotNil(t, sn)
+}
+
+func TestNewSovereignNode_NilNodeShouldError(t *testing.T) {
+	t.Parallel()
+
+	_, err := node.NewSovereignNode(nil)
+	assert.NotNil(t, err)
+}
+
+func TestSovereignNode_GetAllIssuedESDTs(t *testing.T) {
```

Comment:
All these tests below are **not ok**.
Please **do not** duplicate/copy paste code. The base functionality is already being tested in `node_test.go` package.

If you want to test the whole functionality anyway, you could create a base test function for these functions and call it in both `node_test` and `sovereignNode_test.go`.

 However, the thing we want to test in this case(and the reason we have this PR in the first place) is to call this func with a shard coordinator(injected from args) that has `chain id == core.MetachainShardId`. For regular node, if you are a metachain node, this should return `ErrMetachainOnlyEndpoint`, but on sovereign shard should work regardless